boot/bootstate16.go

```diff
@@ -136,14 +136,18 @@ func (s16 *bootState16) markSuccessful(update bootStateUpdate) (bootStateUpdate,
 	env := u16.env
 	toCommit := u16.toCommit
 
+	tryBootVar := fmt.Sprintf("snap_try_%s", s16.varSuffix)
+	bootVar := fmt.Sprintf("snap_%s", s16.varSuffix)
+
 	// snap_mode goes from "" -> "try" -> "trying" -> ""
 	// so if we are not in "trying" mode, nothing to do here
 	if env["snap_mode"] != TryingStatus {
+		// clean the try var anyways in case it was leftover from a rollback,
```

Comment:
does this needs a new unit test?

Reply:
mmm probably yes I will try to write a couple small ones, it's a bit unfortunate that we have a significant delta between test coverage on uc16/uc18 boot logic than we do on uc20 now and it's a bit unclear how we could share a lot of the same code for that.

Reply:
some basic tests added now